raitools/raitools/_managers/explainer_manager.py

```diff
@@ -0,0 +1,135 @@
+        examples (# examples x # features) on which to explain the
+        model's output.
+    :type evaluation_examples: pandas.DataFrame
+    :param target_column: The name of the label column.
+    :type target_column: str
+    :param classes: Class names as a list of strings.
+        The order of the class names should match that of the model
+        output.  Only required if explaining classifier.
+    :type classes: list
+    """
+
+    def __init__(self, model, initialization_examples, evaluation_examples,
+                 target_column, classes=None):
+        """Defines the ExplainerManager for explaining a model.
+
+        :param model: The model to explain.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param initialization_examples: A matrix of feature vector
+            examples (# examples x # features) for initializing the explainer.
+        :type initialization_examples: pandas.DataFrame
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param classes: Class names as a list of strings.
+            The order of the class names should match that of the model
+            output.  Only required if explaining classifier.
+        :type classes: list
+        """
+        self._model = model
+        self._initialization_examples = \
+            initialization_examples.drop(columns=[target_column])
+        self._evaluation_examples = \
+            evaluation_examples.drop(columns=[target_column])
+        self._is_run = False
+        self._is_added = False
+        self._surrogate_model = None
+        self._features = list(self._initialization_examples.columns)
+        self._classes = classes
+        self._target_column = target_column
+        self._explanation = None
+
+    def add(self):
+        """Add an explainer to be computed later."""
+        if self._is_added:
+            warnings.warn(("Ignoring.  Explanation has already been added, "
+                           "currently limited to one explainer type."))
+            return
+        is_sparse = issparse(self._evaluation_examples)
+        num_cols = self._evaluation_examples.shape[1]
+        many_cols = num_cols > SparseNumFeaturesThreshold
+        self._surrogate_model = LGBMExplainableModel
+        if is_sparse and many_cols:
+            self._surrogate_model = LinearExplainableModel
+        self._is_added = True
+
+    def compute(self):
+        """Creates an explanation by running the explainer on the model."""
+        if self._is_run:
+            return
+        model_task = ModelTask.Unknown
+        explainer = MimicExplainer(self._model,
```

Comment:
Also, do the explainers also embed copies of the data? I've never dug into the details of the structure of the returned objects. If so, then we ought to delete it from this copy, and refer to the centrally held data.

Reply:
The explanation has a copy of the dataset, and the explainer does too.  But, I'm not sure why we would want to delete the dataset from the explanation?  Is that just to reduce memory usage?

Reply:
RAIAnalyzer won't be modifying the input dataset internally, will it?

Reply:
Well, yes - to reduce memory usage. On the fairness side, I was expecting that we would only store the metrics, and not the copies of the `y_true` `y_pred` and `sensitive_feature` arrays. Those will all add up, for no good reason. Then the dashboard wanted the particular arrays, it could look them up in the central store in the Analyser object.